### PR TITLE
Do not send body when HTTP verb is DELETE

### PR DESCRIPTION
**Pull request by st**
See https://tickets.puppetlabs.com/browse/TK-66
Connected to #567

### CODE REVIEW

File: ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj [hunk 1]
```diff
@@ -43,7 +44,7 @@
 
 (defn- slurp-body-binary
   [request]
-  (if-let [len (when (:body request) (get-in request [:headers "content-length"]))]
+    (if-let [len (when (and (not= :delete (:request-method request)) (:body request)) (get-in request [:headers "content-length"]))]
```
Comment (konstan) [inline ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj]: Maybe like this?
```
(defn delete-req?
  [request]
  (= :delete (:request-method request)))

 (defn- slurp-body-binary
    [request]
    (if-not (delete-req? request)
      (if-let [len (when (:body request) (get-in request [:headers "content-length"]))]
        ...

```

Comment (st) [inline ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj]: See refactor.

File: ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj [hunk 2]
```diff
@@ -41,9 +42,23 @@
...
 
-(defn- slurp-body-binary
+(defn- not-delete?
   [request]
-  (if-let [len (when (:body request) (get-in request [:headers "content-length"]))]
+  (not= :delete (:request-method request)))
+
+(defn- body?
+  [request]
+  (:body request))
```
Comment (konstan) [inline ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj]: This `body?` is not a predicate in the canonical sense.  Maybe better make it to return boolean.  Something like
```
(not (nil? (:body request)))
```

Comment (st) [inline ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj]: You're right.
Can you do the change and merge?

Comment (loomis) [inline ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj]: My 2 cents.  The two expressions `(:body request)` and `(body? request)` seem equally clear and for all intents and purposes have the same effect.  Is `body?` needed at all?

Comment (st) [inline ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj]: I added this function to be symmetric with first check (not-delete?).
It seemed clearer to me.

**Pull request merged**